### PR TITLE
New make targets for downloading and importing SQL dump file

### PR DESCRIPTION
**Pull request by mabe13**
Pull Request will add the follow actions as make targets:
- download sql dump
- create db docker image
- create ecr repository
- push to ecr
- add addition make targets for test loading the database dump locally

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@
 .python-version
 .vscode
 *.code-workspace
+*.sql.gz
+dos-dump.sql.gz
```
Comment (stefaniuk) [inline .gitignore:7]: This line `dos-dump.sql.gz` is not needed. Please, can we you include an empty line at the end of file?

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,5 @@
 .python-version
 .vscode
 *.code-workspace
+*.sql.gz
```
Comment (stefaniuk) [inline .gitignore:6]: This line `*.sql.gz` should be l=placed in `build/docker/data/assets/sql/.gitignore` file. Therefore, please can we remove it from here?

File: Makefile [hunk 3]
```diff
@@ -19,15 +19,36 @@ create-instance: ## Creates RDS Instance - mandatory: INSTANCE_NAME=[name];
 	echo -e "\nPassword Secret Endpoint:"
 	echo $(INSTANCE_NAME)-nonprod-dos_db_password
 
-download-sql-dump:
+download-sql-dump: ## Downloads the latest DoS database dump and gunzips it
```
Comment (stefaniuk) [inline Makefile:22]: Can we use a single `#` character for comments for the project flow targets?

File: Makefile [hunk 4]
```diff
@@ -19,15 +19,36 @@ create-instance: ## Creates RDS Instance - mandatory: INSTANCE_NAME=[name];
...
 	# TODO: Downlaod the latest DoS database SQL dump file
```
Comment (stefaniuk) [inline Makefile:23]: Can we remove unnecessary comments, please?

File: Makefile [hunk 5]
```diff
@@ -19,15 +19,36 @@ create-instance: ## Creates RDS Instance - mandatory: INSTANCE_NAME=[name];
...
 	# INPUT:
 	#		- link to an SQL dump file
+	make aws-s3-download URI=nhsd-texasplatform-service-dos-lk8s-nonprod/\
```
Comment (stefaniuk) [inline Makefile:26]: Can we refactor this code block to something similar which first searches for the file then downloads it:
```
	file=$(shell aws s3 ls s3://nhsd-texasplatform-service-dos-lk8s-nonprod | sort | grep "dos-pg-dump-.*-clean-PU.sql.gz" | tail -n 1 | awk '{ print $$4 }')
	make aws-s3-download \
		URI=nhsd-texasplatform-service-dos-lk8s-nonprod/$$file \
		FILE=/project/build/docker/data/assets/sql/dos-dump.sql.gz
```

File: Makefile [hunk 6]
```diff
@@ -19,15 +19,36 @@ create-instance: ## Creates RDS Instance - mandatory: INSTANCE_NAME=[name];
...
+	$(shell aws s3 ls s3://nhsd-texasplatform-service-dos-lk8s-nonprod | \
```
Comment (stefaniuk) [inline Makefile:27]: Can we use indentations in new lines after `\`, please?

File: Makefile [hunk 7]
```diff
@@ -19,15 +19,36 @@ create-instance: ## Creates RDS Instance - mandatory: INSTANCE_NAME=[name];
...
+	sort | grep "dos-pg-dump-.*-clean-PU.sql.gz" | tail -n 1 | \
+	awk '{print $$4}') \
+	FILE=/project/build/docker/data/assets/sql/dos-dump.sql.gz
+	gzip -d build/docker/data/assets/sql/dos-dump.sql
```
Comment (stefaniuk) [inline Makefile:31]: Can we convert `build/docker/data/assets/sql/dos-dump.sql` to `$(DOCKER_DIR)/data/assets/sql/dos-dump.sql`?

File: Makefile [hunk 8]
```diff
@@ -19,15 +19,36 @@ create-instance: ## Creates RDS Instance - mandatory: INSTANCE_NAME=[name];
...
+
+push-dos-database-image: ## Pushes the database image to the ECR repository
+	make docker-login
+	make docker-push NAME=data
 
 populate-database:
 	# TODO: Deploy k8s job to run the scripts agains the RDS instance
 	#		- instance name / location
 
+help: ## Show all make targets
```
Comment (stefaniuk) [inline Makefile:48]: No need to include `help` this is already implemented in the `build/automation/init.mk` file.

File: Makefile [hunk 9]
```diff
@@ -19,15 +19,36 @@ create-instance: ## Creates RDS Instance - mandatory: INSTANCE_NAME=[name];
...
+	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//'
+
 # ==============================================================================
 
 .SILENT:
+.DEFAULT_GOAL := help
```
Comment (stefaniuk) [inline Makefile:54]: No need to set a default goal, this is already implemented in the `build/automation/init.mk` file.

File: build/docker/.gitignore [hunk 10]
```diff
@@ -0,0 +1,4 @@
+*.sql.gz
```
Comment (stefaniuk) [inline build/docker/.gitignore:1]: Can we move this tow SQL ignore lines to `build/docker/data/assets/sql/.gitignore` file?

File: .gitignore (hunk 2 above)
Comment (JackPlowman) [reply .gitignore]: Matt has added it in his branch and I have removed it from this gitignore

File: Makefile [hunk 11]
```diff
@@ -1,12 +1,11 @@
 PROJECT_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
 include $(abspath $(PROJECT_DIR)/build/automation/init.mk)
 
+DOS_DB_FILE=$(shell aws s3 ls s3://nhsd-texasplatform-service-dos-lk8s-nonprod | sort | grep "dos-pg-dump-.*-clean-PU.sql.gz" | tail -n 1 | awk '{ print $$4 }')
```
Comment (stefaniuk) [inline Makefile:4]: This expression is evaluated every time we run Makefile functionality but only used in the `download-sql-dump`. Please, can we move it to that target?

File: build/docker/data/assets/script/load_dos_db.sh [hunk 12]
```diff
@@ -0,0 +1,7 @@
+#! /bin/sh.
```
Comment (stefaniuk) [inline build/docker/data/assets/script/load_dos_db.sh:1]: Can we place that file in the `build/docker/data/assets/sbin/init.d/` directory and rename it to `00-load-database.sh`?

Comment (stefaniuk) [inline build/docker/data/assets/script/load_dos_db.sh:1]: This is wrong `#! /bin/sh.` it should be `#!/bin/bash`

File: build/docker/data/assets/script/load_dos_db.sh [hunk 13]
```diff
@@ -0,0 +1,7 @@
...
+
+# Extracting DoS database
+gzip -d $SQL_DIR/dos-dump.sql.gz
+
+# PSQL to load DoS Database on to test database
+psql -f $SQL_DIR/dos-dump.sql --host $PGHOST --port $PGPORT --username $PGUSERNAME --dbname $PGDATABASENAME
```
Comment (stefaniuk) [inline build/docker/data/assets/script/load_dos_db.sh:7]: Please remove line 6 and 7 as the `psql` command will start automatically.

File: build/docker/data/assets/script/load_dos_db.sh [hunk 14]
```diff
@@ -0,0 +1,7 @@
+#! /bin/sh.
+
+# Extracting DoS database
```
Comment (stefaniuk) [inline build/docker/data/assets/script/load_dos_db.sh:3]: Why do we need that comment? What value does it add?

File: build/docker/docker-compose-test.yml [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+      POSTGRES_DB: postgres
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+      PGPASSWORD: postgres
+      PGHOST: test-db.$PROJECT_NAME_SHORT.local
+      PGPORT: 5432
+      PGUSERNAME: postgres
+      PGDATABASENAME: postgres
+      SQL_DIR: /sql
+    command: ["sh", "/script/load_dos_db.sh"]
```
Comment (stefaniuk) [inline build/docker/docker-compose-test.yml:37]: Please, convert that line to `command: ["sql"]` which will execute any SQL script in the `/sql` directory.

File: build/docker/docker-compose-test.yml [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+        aliases:
+          - test-db.$PROJECT_NAME_SHORT.local
+
+  data:
+    image: $DOCKER_REGISTRY/data:latest
+    container_name: data
+    hostname: data
+    environment:
+      PROFILE: $PROFILE
+      POSTGRES_DB: postgres
```
Comment (stefaniuk) [inline build/docker/docker-compose-test.yml:28]: Please, can we use the following variable names:
```
DB_HOST
DB_PORT
DB_NAME
DB_USERNAME
DB_PASSWORD
```
They are already defined and used in the image.

**Pull request merged**